packages/components/src/components/segment/segment.tsx

```diff
@@ -80,6 +83,11 @@ export class Segment {
 
   private focusableElement: HTMLElement;
 
+  @Watch('selected')
+  selectionChanged() {
+    emitEvent(this, 'scaleSelectionChanged');
+  }
+
```

Comment:
IMO the `watch` approach seems promising, and it would be much better to have the  `scaleClick` event emit here on change of the 'selected' state.  This makes the click and programmatic change of the selection land in the same place.